pkg/agent/controller/traceflow/traceflow_controller.go

```diff
@@ -491,6 +492,14 @@ func (c *Controller) preparePacket(tf *crdv1beta1.Traceflow, intf *interfacestor
 		} else if packet.DestinationIP.To4() != nil {
 			return nil, errors.New("destination Service does not have an IPv6 ClusterIP")
 		}
+		switch dstSvc.Spec.Ports[0].Protocol {
+		case corev1.ProtocolTCP:
+			packet.IPProto = protocol.Type_TCP
+			packet.TCPFlags = uint8(2)
+		case corev1.ProtocolUDP:
+			packet.IPProto = protocol.Type_UDP
+		}
+		packet.DestinationPort = uint16(dstSvc.Spec.Ports[0].Port)
```

Comment:
can we actually make this conditional so that it is skipped when `liveTraffic` is true?

Reply:
Why we want to skip it incase of liveTraffic ?
It can happen that liveTraffic is enabled with source as a pod and destination as a service but no protocol/port is provided.

Reply:
Because for live traceflow, we may want to capture _any_ Service traffic, regardless of port / protocol (in case the Service has multiple of those). And in this case, it makes sense to only filter based on the destination IP (i.e., the Service ClustrerIP).